octree_web_viewer/src/bin/points_web_viewer.rs

```diff
@@ -12,44 +12,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use octree_web_viewer::utils;
-use point_viewer::octree;
+use octree_web_viewer::backend_error::PointsViewerError;
+use octree_web_viewer::state::AppState;
+use octree_web_viewer::utils::start_octree_server;
 use std::path::PathBuf;
 use std::sync::Arc;
 use structopt::StructOpt;
 
 /// HTTP web viewer for 3d points stored in OnDiskOctrees
 #[derive(StructOpt, Debug)]
 #[structopt(name = "points_web_viewer", about = "Visualizing points")]
-struct CommandlineArguments {
+pub struct CommandLineArguments {
     /// The octree directory to serve, including a trailing slash.
-    octree_path: String,
+    /// this overrides the path_* options
+    #[structopt(name = "DIR")]
```

Comment:
I find the current way to specify either a full path or prefix/id/suffix pretty confusing. How about using mutually exclusive arguments (see e.g. https://github.com/TeXitoi/structopt/issues/23#issuecomment-326974584)?

Reply:
done